tests/unit/pages/edit-profile/EditProfile.spec.jsx

```diff
@@ -1,14 +1,102 @@
-import { renderWithProviders } from '~tests/test-utils'
+import { screen, waitFor, fireEvent } from '@testing-library/react'
+import { renderWithProviders, mockAxiosClient } from '~tests/test-utils'
+import { URLs } from '~/constants/request'
 
 import EditProfile from '~/pages/edit-profile/EditProfile'
 
+const userId = '63f5d0ebb'
+const userRole = 'tutor'
+
+const mockState = {
+  appMain: { userId: userId, userRole: userRole }
+}
+
+const userMock = 'UserProfileMock'
+
+vi.mock(
+  '~/containers/edit-profile/profile-general-tab/ProfileGeneralTab',
+  () => ({
+    __esModule: true,
```

Comment:
Please note that we don't need to use `__esModule: true` in the current Vitest version.